pkg/kubelet/kuberuntime/kuberuntime_container_linux.go

```diff
@@ -100,7 +101,7 @@ func (m *kubeGenericRuntimeManager) generateLinuxContainerResources(pod *v1.Pod,
 	if _, cpuRequestExists := container.Resources.Requests[v1.ResourceCPU]; cpuRequestExists {
 		cpuRequest = container.Resources.Requests.Cpu()
 	}
-	lcr := m.calculateLinuxResources(cpuRequest, container.Resources.Limits.Cpu(), container.Resources.Limits.Memory())
+	lcr := m.calculateLinuxResources(cpuRequest, container.Resources.Limits.Cpu(), container.Resources.Limits.Memory(), kubeapiqos.GetPodQOS(pod))
```

Comment:
here we have the pod for which the resources are been computed, so we can call `m.containerManager.HasExclusiveCPUs(pod)` and pass down the flag.

I need to review carefully the flow to check for possible issues but it seems doable.

Reply:
hmm assuming the cpu assignment occurs prior to generateContainerResources being called then yes, will swap out. this also removes need to have `StaticCPUPolicyConditionsSatisfied` defined and called within `calculateLinuxResources`